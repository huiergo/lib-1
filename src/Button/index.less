@import '../style/common.less';



.@{btnPrefix} {
    display: flex;
    justify-content: center;
    align-items: center;
    &-default{
        background-color: #fff;
        color: #000;
        border: 1px solid #333;
    }
    &-highlight{
        background-color: @heima-ui-primary-color;
        color: #fff;
    }
    &-disabled{
        background-color:@heima-ui-disabled-color;
        color: #999;
        border: 1px solid #999;
    }
}