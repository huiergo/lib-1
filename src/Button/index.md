---
nav:
  title: Components
  path: /components

group:
  title: '基础组件'
  order:  2
---

## Button

Button 示例:

### 普通按钮
```tsx
import React from 'react';
import { Button } from 'heima-ui-1';

export default () => <Button text="普通按钮" style={{width:200}}/>;
```

### 高亮按钮
```tsx
import React from 'react';
import { Button } from 'heima-ui-1';

export default () => <Button text="普通按钮" type='highlight' style={{width:200}}/>;
```


### 禁用按钮
```tsx
import React from 'react';
import { Button } from 'heima-ui-1';

export default () => <Button text="普通按钮" type='highlight' disabled style={{width:200}}/>;
```
More skills for writing demo: https://d.umijs.org/guide/basic#write-component-demo
