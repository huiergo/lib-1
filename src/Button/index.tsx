import React, { ReactNode, useContext } from 'react';
import cs from 'classnames';
import './index.less'
import { ConfigContext } from '../ConfigProvider';

interface ButtonProps {
    type?: 'highlight' | 'default',
    text?: string,
    children?: ReactNode,
    disabled?: boolean,
    style?: object,
    className?: string,
    icon?: ReactNode,
    onClick?: React.MouseEventHandler<HTMLDivElement>
}

export default (props: ButtonProps) => {
    const { type = 'default', text, children, disabled, style, className, icon, onClick } = props

    // const prefix = 'heima-ui'

    const { prefix } = useContext(ConfigContext);
    const btnPrefix = prefix + '-btn'

    const classes = cs(btnPrefix, {
        [`${btnPrefix}-${type}`]: type,
        [`${btnPrefix}-disabled`]: disabled
    }, className)



    return (
        <div className={classes} style={style} onClick={(e) => {
            if (disabled) {
                return
            }
            onClick && onClick(e)
        }}>
            {icon}
            {text}
            {children}
        </div>
    )
};
