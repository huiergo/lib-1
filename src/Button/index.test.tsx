import '@testing-library/jest-dom';
import React from 'react';
import Enzyme, { render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Button from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Button 组件', () => {
  it('renders the text by render', () => {
    const text = '按钮内容'
    const wrapper = render(<Button text={text}></Button>)
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.text()).toEqual(text)
  })
  /**
* 测试： button 正常点击
*/
  it('onClick', () => {
    const onclick = jest.fn()
    const wrapper = shallow(<Button onClick={onclick}>可点按钮</Button>)
    wrapper.simulate('click')
    expect(onclick).toBeCalled()
  })

  /**
* 测试： button 正常点击
*/
  it('onClick disabled', () => {
    const onclick = jest.fn()
    const wrapper = shallow(<Button onClick={onclick} disabled>禁用按钮</Button>)
    wrapper.simulate('click')
    expect(onclick).not.toBeCalled()
  })


});
