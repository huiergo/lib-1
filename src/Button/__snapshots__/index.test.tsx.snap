// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 组件 renders the text by render 1`] = `
<div
  class="heima-ui-btn heima-ui-btn-default"
>
  按钮内容
</div>
`;
